### PR TITLE
[BUG][typescript-fetch] wrong response for simple types

### PR DESCRIPTION
**Pull request by Serranya**
See #9364
See #2870

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @davidgamero @mkusaka

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache [hunk 1]
```diff
@@ -306,7 +306,11 @@ export class {{classname}} extends runtime.BaseAPI {
         return new runtime.JSONApiResponse<any>(response);
         {{/isArray}}
         {{#returnSimpleType}}
-        return new runtime.TextApiResponse(response) as any;
+        if (response.headers.get('content-type') === 'application/json') {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache:309]: can you introduce a helper function, similar to https://github.com/OpenAPITools/openapi-generator/blob/43493a94fef55e874391760053066c1cc5efd6bb/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache#L179?

Comment (Serranya) [reply modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache:309]: Sure. I will copy the logic you linked.

Comment (Serranya) [reply modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache]: I used the regex you linked but changed the capturing groups to non-capturing groups.

**Comment by Serranya** [comment]
@macjohnny can you take another look please?

**Review by macjohnny — APPROVED** [review]
LGTM

can you please re-generate the samples?

**Comment by Serranya** [comment]
@macjohnny I updated the sample files.

**Pull request merged**